core/src/main/java/io/grpc/ManagedChannelProvider.java

```diff
@@ -87,6 +87,27 @@ public static ManagedChannelProvider provider() {
     return provider;
   }
 
+  private static ClassLoader getCorrectClassLoader() {
+    if (isAndroid()) {
+      // When android:sharedUserId or android:process is used, Android will setup a dummy
+      // ClassLoader for the thread context (http://stackoverflow.com/questions/13407006),
+      // instead of letting users to manually set context class loader, we choose the
+      // correct class loader here.
+      return ManagedChannelProvider.class.getClassLoader();
```

Comment:
@carl-mastrangelo, no, the dummy class loader just output a warning to logcat and will not load anything.

Yes, as [Class.forName](http://docs.oracle.com/javase/7/docs/api/java/lang/Class.html#forName(java.lang.String)) says:
```
Returns the Class object associated with the class or interface with the given string name. Invoking this method is equivalent to:
Class.forName(className, true, currentLoader)
where currentLoader denotes the defining class loader of the current class.
```